client/my-sites/plan-features-2023-grid/components/storage-add-on-dropdown.tsx

```diff
@@ -92,9 +94,10 @@ export const StorageAddOnDropdown = ( {
 			label={ label }
 			options={ selectControlOptions }
 			value={ selectedOption }
-			onChange={ ( { selectedItem }: { selectedItem: { key: WPComStorageAddOnSlug } } ) =>
-				setSelectedStorageOptionForPlan( { addOnSlug: selectedItem?.key || '', planSlug } )
-			}
+			onChange={ ( { selectedItem }: { selectedItem: { key: WPComStorageAddOnSlug } } ) => {
+				onStorageAddOnClick && onStorageAddOnClick();
+				setSelectedStorageOptionForPlan( { addOnSlug: selectedItem?.key || '', planSlug } );
```

Comment:
Under what conditions would an empty `''` `addOnSlug` make sense?

Reply:
You're right 👍 I've updated the logic such that the data store storage option action is only dispatched when there _is_ a meaningful key. Thank you!